siad/main.go

```diff
@@ -87,10 +82,28 @@ func init() {
 }
 
 func startEnvironment(*cobra.Command, []string) {
-	if err := config.expand(); err != nil {
+	daemonConfig := DaemonConfig{
+		APIAddr:     config.Siad.APIaddr,
+		RPCAddr:     config.Siacore.RPCaddr,
+		NoBootstrap: config.Siacore.NoBootstrap,
+
+		HostDir: config.Siacore.HostDirectory,
+
+		Threads: 1,
+
+		DownloadDir: config.Siad.DownloadDirectory,
+
+		WalletDir: config.Siad.WalletFile,
+	}
+	err := config.expand()
+	if err != nil {
 		fmt.Println("Bad config value:", err)
-	} else if err := startDaemon(config); err != nil {
+	}
+	_, err = newDaemon(daemonConfig)
+	if err != nil {
 		fmt.Println("Failed to start daemon:", err)
+	} else {
+		select {}
```

Comment:
I thought the error convention was:
```go
_, err = newDaemon(daemonConfig)
if err != nil {
    fmt.Println("Failed to start daemon:", err)
    return
}

// block indefinitely
select {}
```

I don't like `select {}` though. How will the program exit? I'd prefer something like:

```go
d, err = newDaemon(daemonConfig) // doesn't spawn handler process
if err != nil {
    fmt.Println("Failed to start daemon:", err)
    return
}

// handle requests until quit signal is received
d.listen()
```